src/backend/columnar/columnar_customscan.c

```diff
@@ -1540,8 +1561,10 @@ ColumnarScan_ReScanCustomScan(CustomScanState *node)
 	ColumnarScanState *columnarScanState = (ColumnarScanState *) node;
 
 	ResetExprContext(columnarScanState->css_RuntimeContext);
+	List *allClauses = list_concat(list_copy(linitial(cscan->custom_exprs)),
```

Comment:
could the first list ever be `NIL`? This would cause a reuse of the pointer for the second list, which might be 
problematic based on the use after this place.

Would it make sense to also make a copy of the list stored in the second element? Or do we know that the `allClauses` list will not be mutated?

Reply:
I changed this so that the second list is all the usable clauses, and the plain clauses are a subset. 

expression_tree_mutator() is generally supposed to return a modified copy; it's comment reads: "The original tree is (normally) not changed.". So I think we're OK without an additional copy.

I don't know if this entirely addresses your concern though. Can you restate after looking at the newest version?